Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -363,6 +363,31 @@ class UserScriptManager {
       forMainFrameOnly: true,
       in: .defaultClient)
   }()
+  
+  private let ReadyStateScript: WKUserScript? = {
+    guard let path = Bundle.main.path(forResource: "ReadyState", ofType: "js"), let source = try? String(contentsOfFile: path) else {
+      log.error("Failed to load ReadyState.js")
+      return nil
+    }
+
+    var alteredSource = source
+    let token = UserScriptManager.securityTokenString
+
+    let replacements = [
+      "$<security_token>": token,
+      "$<handler>": "ReadyState_\(messageHandlerTokenString)",
```

Comment:
what happens if a webpage includes some JS which includes the string `$<security_token>`? will it get replaced and potentially allow that page to see the token?

Reply:
Web pages will not be able to do that because this is injected `atDocumentStart` so before any of the web page's source code is executed.